components/collective-page/index.js

```diff
@@ -299,6 +317,29 @@ class CollectivePage extends Component {
         css={collective.isArchived ? 'filter: grayscale(100%);' : undefined}
         data-cy="collective-page-main"
       >
+        {notification && (
+          <TemporaryNotification onDismiss={this.dismissNotification} type={notification.type}>
+            {notification.type === 'error' ? (
+              <FormattedMessage
+                id="ApplyToHost.error"
+                defaultMessage="An error occurred while applying to {hostName} with {collectiveName}."
+                values={{
+                  hostName: collective.name,
+                  collectiveName: notification.message,
+                }}
+              />
+            ) : (
+              <FormattedMessage
+                id="ApplyToHost.success"
+                defaultMessage="{collectiveName} has applied to be hosted by {hostName}."
+                values={{
+                  hostName: collective.name,
+                  collectiveName: notification.message,
+                }}
+              />
```

Comment:
The goal of https://github.com/opencollective/opencollective-frontend/pull/5355#discussion_r522193697 was to have a generic system for messages, but here we still have specific code for applications. Why not just:

```jsx
{notification && (
  <TemporaryNotification onDismiss={this.dismissNotification} type={notification.type}>
    {notification.message}
  </TemporaryNotification>
)}
```

Then you can generate the message from the place that really knows about what it's related to (`CollectiveNavbarActionsMenu.js`).